src/app/core/layers/map/vectorlayer.js

```diff
@@ -86,14 +86,14 @@ proto._makeStyle = function(styleConfig) {
 };
 
 proto.getFeatures = function(options={}) {
-  const d = $.Deferred();
-  this.provider.getFeatures(options)
-    .then(features => {
-      this.addFeatures(features);
-      d.resolve(features);
-    })
-    .fail(err => d.reject(err));
-  return d.promise()
+  return new Promise((resolve, reject) =>{
+    this.provider.getFeatures(options)
+      .then(features => {
+        this.addFeatures(features);
+        resolve(features);
+      })
+      .catch(err => reject(err));
```

Comment:
double check, isn't this a superfluous statement?

```js
.catch(err => reject(err));
```

Reply:
> ```js
> .catch(err => reject(err));
> ```

I didn't mark them all, but I found this kind of repetition quite often in the code